### PR TITLE
bridge_ui: Added WalletConnect support for Ethereum

### PR DESCRIPTION
**Pull request by kev1n-peters**
This change is [Reviewable](https://reviewable.io/reviews/certusone/wormhole/1155)

### CODE REVIEW

File: bridge_ui/src/contexts/EthereumProviderContext.tsx [hunk 1]
```diff
@@ -42,91 +58,201 @@ export const EthereumProviderProvider = ({
...
+          });
         }
-      })
-      .catch(() => {
-        setProviderError("Please install MetaMask");
-      });
+      } catch (error) {
+        console.error(error);
+      }
+      if (process.env.REACT_APP_INFURA_API_KEY) {
```
Comment (kev1n-peters) [inline bridge_ui/src/contexts/EthereumProviderContext.tsx:81]: Maybe we should set this for Netlify builds?

**Review by panoel — DISMISSED** [review]

File: bridge_ui/src/hooks/useIsWalletReady.ts [hunk 2]
```diff
@@ -56,7 +64,18 @@ function useIsWalletReady(
...
+        EVM_RPC_MAP[correctEvmNetwork] === undefined
+      ) {
+        // WalletConnect requires a rpc url for this chain
+        // Force user to switch connect type
+        disconnect();
+        return;
+      }
+
       try {
+        console.log("switching");
```
Comment (panoel) [inline bridge_ui/src/hooks/useIsWalletReady.ts:78]: Not sure where we are on log messages.  Doesn't bother me.  But a one word log may not provide much context.

**Review by panoel — APPROVED** [review]

**Review by evan-gray — COMMENTED** [review]
Tested with Solana, Avalanche, and Goerli on testnet and seems to be working well! One thing I noticed was that events in MetaMask still fired even after it was disconnected (and swapped for WalletConnect) so we should probably remove all listeners on disconnect.

File: bridge_ui/src/hooks/useIsWalletReady.ts (hunk 2 above)
Comment (evan-gray) [reply bridge_ui/src/hooks/useIsWalletReady.ts:78]: agree, we should probably remove the log

**Review by evan-gray — APPROVED** [review]

**Review by panoel — APPROVED** [review]

**Pull request merged**